folio-spring-system-user/src/main/java/org/folio/spring/service/SystemUserService.java

```diff
@@ -85,17 +78,22 @@ public UserToken authSystemUser(SystemUser user) {
 
   public UserToken authSystemUser(String tenantId, String username, String password) {
     log.info("Attempting to issue token for system user [tenantId={}]", tenantId);
-    try (var fex =
-             new FolioExecutionContextSetter(contextBuilder.forSystemUser(prepareSystemUser(tenantId, username)))) {
+    var systemUser = prepareSystemUser(tenantId, username);
+    try (var fex = new FolioExecutionContextSetter(contextBuilder.forSystemUser(systemUser))) {
       var token = getToken(new UserCredentials(username, password));
       log.info("Token for system user has been issued [tenantId={}]", tenantId);
       return token;
     } catch (Exception exp) {
       log.error("Unexpected error occurred while setting folio context" + exp);
-      throw new AuthorizationException(CANNOT_RETRIEVE_OKAPI_TOKEN_FOR_TENANT + tenantId);
+      throw new SystemUserAuthorizationException(TOKEN_FAILED_MSG + tenantId);
```

Comment:
Minor. Maybe good to consider using String.format(...) like inside other exceptions in this class.

Reply:
Would be an overhead when only one string added to the end